src/journal/test/JournalTest.hs

```diff
@@ -44,36 +45,79 @@ startJournalFake = FakeJournal Vector.empty 0
 
 appendBSFake :: ByteString -> FakeJournal -> (FakeJournal, Either AppendError ())
 appendBSFake bs fj@(FakeJournal bss ix)
-  | unreadBytes < limit = (FakeJournal (Vector.snoc bss bs) ix, Right ())
+  | position < limit =
+    trace (unlines [ "TRACE"
+                   , "ix: " ++ show ix
+                   , "readBytes: " ++ show readBytes
+                   , "unreadBytes: " ++  show unreadBytes
+                   , "readPadBytes: " ++  show readPadBytes
+                   , "unreadPadBytes: " ++ show unreadPadBytes
+                   , "position: " ++ show position
+                   , "limit: " ++ show limit
+                   ])
+    (FakeJournal (Vector.snoc bss bs) ix, Right ())
   | otherwise           = (fj, Left BackPressure)
   where
     termLen :: Int
     termLen = oTermBufferLength testOptions
 
+    position = readBytes + unreadBytes
+
     limit :: Int
-    limit = termLen `div` 2
+    limit = readBytes + termLen `div` 2
+
+    readBytes :: Int
+    readBytes = Vector.sum
+              . Vector.cons readPadBytes
+              . Vector.map (\bs -> hEADER_LENGTH + BS.length bs)
+              $ Vector.slice 0 (max 0 (ix - 1)) bss
+
+    readPadBytes :: Int
+    readPadBytes =
```

Comment:
Hmm so if we need both, maybe we should compute them at the same time?
```haskell
readPadBytes, unreadPadBytes :: Int
(readPadBytes, unreadPadBytes) =
   let (lbss, rbss) = ...
       ...
   in (lpad, rpad)

Reply:
Thanks, fixed!